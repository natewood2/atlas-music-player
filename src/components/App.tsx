//import React from "react";
import Footer from "./Footer";
import MusicPlayer from "./MusicPlayer";

const toggleTheme = () => {
    document.documentElement.classList.toggle("dark");
  };

function App() {
  return (
    <div className="h-full flex flex-col justify-between rounded-lg bg-gray-200 dark:bg-gray-700 p-8 min-h-screen">
      <button 
        onClick={toggleTheme}
        className="absolute top-4 right-4 p-2 rounded-lg bg-gray-200 dark:bg-zinc-600 text-gray-800 dark:text-white"
      >
        Toggle Theme
      </button>
      <div className="flex flex-1 gap-8">
        <div className="flex-1">
          <MusicPlayer />
        </div>
      </div>
      <Footer />
    </div>
  );
}

export default App;
