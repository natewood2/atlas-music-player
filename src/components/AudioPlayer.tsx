import { useEffect, useRef } from 'react';

interface AudioPlayer {
  songUrl?: string;
  isPlaying: boolean;
  volume: number;
  speed: number;
}

export default function AudioPlayer({ songUrl, isPlaying, volume, speed }: AudioPlayer) {
  const audioRef = useRef<HTMLAudioElement | null>(null);

  useEffect(() => {
    if (audioRef.current) {
      if (isPlaying) {
        audioRef.current.play();
      } else {
        audioRef.current.pause();
      }
    }
  }, [isPlaying, songUrl]);

  useEffect(() => {
    if (audioRef.current) {
      audioRef.current.volume = volume;
    }
  }, [volume]);

  useEffect(() => {
    if (audioRef.current) {
      audioRef.current.playbackRate = speed;
    }
  }, [speed]);

  useEffect(() => {
    if (audioRef.current && songUrl) {
      audioRef.current.src = songUrl;
      if (isPlaying) {
        audioRef.current.play();
      }
    }
  }, [songUrl]);

  return (
    <audio 
      ref={audioRef}
      preload="auto"
      controls={false}
    />
  );
}
