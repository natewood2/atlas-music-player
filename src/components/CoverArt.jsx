import React from "react";
import Cover from "/images/cover-art.png";

export default function CoverArt({ src }) {
  return (
    <div className="w-full overflow-hidden rounded-lg bg-gray-200">
      <img 
        src={src || Cover} 
        alt="Album cover" 
        className="h-full w-full object-cover"
      />
    </div>
  );
}