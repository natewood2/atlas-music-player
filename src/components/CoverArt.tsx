import Cover from "/images/cover-art.png";

export default function CoverArt({ src }: { src?: string }) {
  console.log("CoverArt src:", src);
  return (
    <div className="w-full overflow-hidden rounded-lg bg-gray-200">
      <img 
        src={src || Cover} 
        alt="Album cover" 
        className="h-full w-full object-cover"
      />
    </div>
  );
}