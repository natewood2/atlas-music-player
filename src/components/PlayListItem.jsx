import React from 'react';
import SongTitle from './SongTitle';

function formatDuration(seconds) {
  const minutes = Math.floor(seconds / 60);
  const remainingSeconds = seconds % 60;
  return `${minutes}:${remainingSeconds.toString().padStart(2, '0')}`;
}

export default function PlayListItem({ song }) {
  if (!song) return null;

  return (
    <div className="flex justify-between items-center p-2 hover:bg-red-50 rounded">
      <SongTitle song={song} />
      <span className="text-gray-500 dark:text-gray-300 text-sm">{formatDuration(song.duration)}</span>
    </div>
  );
}
