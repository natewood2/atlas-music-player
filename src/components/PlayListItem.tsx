type PlayListItemProps = {
  songName: string;
  songTime: number;
  artist: string;
  isPlaying: boolean;
  onClick: () => void;
}

export function PlayListItem({ songName, songTime, artist, isPlaying, onClick }: PlayListItemProps) {
    const time = songTime.toString();
    const formattedTime = `${time.slice(0, 1)}:${time.slice(1)}`;

    return (
        <div 
            onClick={onClick}
            className={`w-full flex justify-between items-center mb-2 rounded-md cursor-pointer 
                ${isPlaying ? 'bg-secondary dark:bg-bg-color text-white dark:text-secondary' : 'bg-transparent'}`}
        >
            <div className="flex flex-col">
                <p className='font-bold'>{songName}</p>
                <p className='font-bold opacity-70'>{artist}</p>
            </div>
            <p className="font-bold opacity-50">{formattedTime}</p>
        </div>
    );
}