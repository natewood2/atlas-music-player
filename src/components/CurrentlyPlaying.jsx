import React from 'react';
import CoverArt from './CoverArt';
import SongTitle from './SongTitle';
import PlayControls from './PlayControl';
import VolumeControls from './VolumeControl';
import useSongs from '../hooks/useSongs';

export default function CurrentlyPlaying() {
  const songs = useSongs();
  const currentSong = songs[0];

  return (
    <div className="flex flex-col">
      <div className="aspect-square mb-6">
        <CoverArt src={currentSong?.cover} />
      </div>
      
      <div className="space-y-6">
        <SongTitle song={currentSong} />
        <PlayControls />
        <div className="mt-8">
          <VolumeControls />
        </div>
      </div>
    </div>
  );
}
