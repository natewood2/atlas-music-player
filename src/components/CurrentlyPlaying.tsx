import CoverArt from "./CoverArt";
import SongTitle from "./SongTitle";
import PlayControls from "./PlayControl";
import VolumeControls from "./VolumeControl";
import AudioPlayer from "./AudioPlayer";
import { Song } from "./MusicPlayer";

interface CurrentlyPlaying {
  loading: boolean;
  playlist: Song[];
  currentSong: number;
  setCurrentSong: (index: number) => void;
  setIsPlaying: (isPlaying: boolean) => void;
  isPlaying: boolean;
  setVolume: (volume: number) => void;
  setSpeed: (speed: number) => void;
  speed: number;
  volume: number;
}

export default function CurrentlyPlaying({ 
  loading, 
  playlist, 
  currentSong, 
  setCurrentSong, 
  setIsPlaying, 
  isPlaying, 
  setVolume,
  volume,
  setSpeed, 
  speed 
}: CurrentlyPlaying) {
  const currentSongData = playlist[currentSong];
  console.log("Song:", currentSongData);
  console.log("Loadin:", loading);
  console.log("Playlist:", playlist);
  console.log("index of song", currentSong);

  return (
    <div className="flex flex-col w-full m-auto p-5 md:border-r-3 border-secondary dark:border-bg-color">
      <AudioPlayer
        songUrl={currentSongData?.song}
        isPlaying={isPlaying}
        volume={volume}
        speed={speed}
      />
      <CoverArt 
        src={loading ? undefined : currentSongData?.cover} 
      />
      <SongTitle 
        song={loading ? undefined : {
          title: currentSongData?.title || "",
          artist: currentSongData?.artist || ""
        }}
      />
      <PlayControls
        onNext={() => setCurrentSong((currentSong + 1) % playlist.length)}
        onPrevious={() => setCurrentSong((currentSong - 1 + playlist.length) % playlist.length)}
        onPlayPause={() => setIsPlaying(!isPlaying)}
        isPlaying={isPlaying}
        speed={speed}
        setSpeed={setSpeed}
      />
      <VolumeControls setVolume={setVolume} />
    </div>
  );
}