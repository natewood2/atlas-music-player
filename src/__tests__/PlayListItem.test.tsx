import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { PlayListItem } from '../components/PlayListItem';

it('renders PlayListItem ＾ዋ⋏ዋ＾', () => {
  const { container } = render(
    <PlayListItem 
      songName="Test Song"
      songTime={180}
      artist="Test Artist"
      isPlaying={false}
      onClick={() => {}}
    />
  );
  expect(container).toMatchSnapshot();
});