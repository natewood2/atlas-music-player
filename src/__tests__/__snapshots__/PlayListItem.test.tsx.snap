// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders PlayListItem ＾ዋ⋏ዋ＾ 1`] = `
<div>
  <div
    class="w-full flex justify-between items-center mb-2 rounded-md cursor-pointer 
                bg-transparent"
  >
    <div
      class="flex flex-col"
    >
      <p
        class="font-bold"
      >
        Test Song
      </p>
      <p
        class="font-bold opacity-70"
      >
        Test Artist
      </p>
    </div>
    <p
      class="font-bold opacity-50"
    >
      1:80
    </p>
  </div>
</div>
`;
